subsys/bluetooth/host/smp.c

```diff
@@ -4266,6 +4266,16 @@ static int bt_smp_recv(struct bt_l2cap_chan *chan, struct net_buf *buf)
 		return 0;
 	}
 
+	/*
+	 * Bluetooth Core Specification Version 5.2, Vol 3, Part H, page 1667:
+	 * If a packet is received with a Code that is reserved for future use
+	 * it shall be ignored.
+	 */
+	if (hdr->code >= BT_SMP_CMD_RESERVED) {
```

Comment:
This looks like it is handling parts of the if-statement below, so that could be updated.
In fact we could just re-use that check and not use the CMD_RESERVED, that would make it easier to add new handlers as we don't have to adjust the CMD_RESERVED.

`if (hdr->code >= ARRAY_SIZE(handlers))`

Reply:
I did not do that because of the keypress notification: we dont set the keypress bit, and then we abort pairing upon receiving one since no handler is defined. I don't know how zephyr wants to handle this, keep as-is or ignore. If we ignore it I can modify the check below and remove this one

Reply:
Thanks for explaining, I did not notice the missing 0xe in between there.
I checked it up in the specification and it says that it shall be ignored.

```
The keypress field is a 1-bit flag that is used only in the Passkey Entry
protocol and shall be ignored in other protocols. When both sides set that
field to one ....
```

Although I'd prefer to ignore it more explicitly, instead of as an addition to the reserved SMP codes.
So I would prefer to add the handler for the missing SMP keypress PDUs  which simply ignores them
And do that in a commit preceding this commit, that would be cleanest IMO.